app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -204,7 +206,7 @@ public static Intent createNoInfoIntent() {
     public static void populateInfoIntent(Intent intent, ScanData scanData, @NonNull Optional<String> filePath) {
         intent.putExtra(KEY_SEND_INFO_NAME, scanData.getPokemonName());
         intent.putExtra(KEY_SEND_INFO_TYPE, scanData.getPokemonType());
-        intent.putExtra(KEY_SEND_INFO_CANDY, scanData.getCandyName());
+        intent.putExtra(KEY_SEND_INFO_CANDY, scanData.getCandyNames().toArray());
```

Comment:
Can getCandyNames return null?

Reply:
Under current logic flows, no, but I'll add some protection when initializing a new ScanData just in case